shared/driver/driver.go

```diff
@@ -20,18 +20,24 @@ var (
 	_ driver.Connector = &Connector{}
 )
 
+const (
+	DBTypeMaster  = "master"
+	DBTypeReplica = "replica"
+)
+
 // Connector is the DB connector which is used to
 // communicate with the DB API.
 type Connector struct {
-	api plugin.Driver
+	api    plugin.Driver
+	dbType string
 }
 
-func NewConnector(api plugin.Driver) *Connector {
-	return &Connector{api: api}
+func NewConnector(api plugin.Driver, dbType string) *Connector {
+	return &Connector{api: api, dbType: dbType}
```

Comment:
Purely thinking out loud here: if we know the type is (and will be) a binary decision why don't we explicitly make it so by using a boolean instead?

Reply:
0/5. I am not sure if it "always will be" a binary decision. Just keeping space for future. But if you prefer to keep it boolean and then change to string later, I can do that.

Reply:
I think I am not entirely convinced by the naming as I implicitly identify the type of a database as something else entirely and I guess changing to boolean would avoid this issue in the first place. But I'll defer to @isacikgoz and see what he thinks as well before we make any change.

Reply:
Completely 0/5 here. Let me know what you guys think is the best.

Reply:
0/5 on either but if I was implementing this I'd pick a boolean for this 😅

Reply:
Boolean it is then.